04.class/lib/memo_app.js

```diff
@@ -0,0 +1,86 @@
+import { Memo } from "./memo.js";
+import { View } from "./view.js";
+
+export class MemoApp {
+  #view = null;
+
+  constructor() {
+    this.#view = new View();
+  }
+
+  async run(params) {
+    await Memo.createTable();
+
+    if (params.l) {
+      await this.#printMemos();
+    } else if (params.r) {
+      await this.#printMemo();
+    } else if (params.d) {
+      await this.#deleteMemo();
+    } else {
+      await this.#createMemo();
+    }
+
+    await Memo.close();
+  }
+
+  async #printMemos() {
+    const memos = await Memo.all();
+    if (memos.length === 0) {
+      this.#view.printNoMemos();
+    } else {
+      this.#view.printMemos(memos);
```

Comment:
```suggestion
    if (memos.length) {
      this.#view.printMemos(memos);
    } else {
      this.#view.printNoMemos();
```

0が偽であることを利用してこういう書き方をする人も時々見かけます。参考までに。
（ただ、if 正常系 else 異常系 の順で並んだ方が読みやすいのはたしか）

Reply:
個人的には`if (memos.length)`のように省略する派だったのですが、以前に条件は明示した方が良いと指摘されたので明示するようにしていました。
以下その時のコメントです。
https://github.com/ryufuta/js-practices/pull/2#discussion_r1856136324

明示するにしてもご指摘の通り、正常系、異常系の順の方が読みやすいので`if (memos.length) > 0`の方が良いですね💦
修正します。

Reply:
> 以前に条件は明示した方が良いと指摘された

そうですね、ここは好き嫌いが分かれるところですし、僕自身も`if (memos.length)`が好きかと言われるとそうではないので、`> 0`みたいな書き方で全然大丈夫です。
僕の前回のコメントは「直してほしい」というより、「こういうコードも読めるようになっておきましょう」の意図でした。
（もともと省略する派だったらあまり関係ありませんが😅）